### PR TITLE
Add dark/light theme option

### PR DESCRIPTION
**Pull request by krt16s**
This sets the default theme to night mode, which is more coherent with previous versions. It adds a preference activity to toggle between light and dark theme.

The preference activity was missing the actionbar at first. I read about 20 different tutorials on what to do to enable it but all of them seem to be ridiculously  big changes for such a common task. For now I settled with one, but I don't know if that's the way it is supposed to work. I feel like I am mixing different concepts of android design. Also note that this might break on lower APIs, so this should be tested before releasing another version.

### CODE REVIEW

**Comment by dschuermann** [comment]
Regarding preference: Can you copy https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/compatibility/AppCompatPreferenceActivity.java into offline calendar, update to the latest support lib and then use the method from here: https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SettingsActivity.java#L98
This works reasonably well.

File: Offline-Calendar/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -45,7 +46,8 @@
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
-        android:theme="@style/Theme.AppCompat.Light.NoActionBar">
+        android:theme="@style/Theme.AppCompat.NoActionBar">
```
Comment (dschuermann) [inline Offline-Calendar/src/main/AndroidManifest.xml]: I would prefer to keep the Light mode as default.

**Comment by dschuermann** [comment]
Everything else looks good!

File: Offline-Calendar/src/main/AndroidManifest.xml (hunk 1 above)
Comment (krt16s) [inline Offline-Calendar/src/main/AndroidManifest.xml]: Is there a way to stop the flickering when starting an app with one theme set in the manifest and another one set via java?

**Comment by krt16s** [comment]
Pulling in those changes adds quite a lot of dependencies, which I most likely fail to setup correctly. https://github.com/SufficientlySecure/offline-calendar/pull/54/commits/ef7c5f0ae781b23a2b51b885ae992828f13dc567 builds, but fails with  

`03-21 18:37:24.460  1604  1604 E AndroidRuntime: Caused by: android.view.InflateException: Binary XML file line #23: addView(View, LayoutParams) is not supported in AdapterView
`

**Comment by dschuermann** [comment]
try gradle clean

**Comment by krt16s** [comment]
Doesn't change anything.

**Comment by krt16s** [comment]
My quickfix until I (might) have some time again would be to disable the preference screen on older devices, like in https://github.com/krt16s/offline-calendar/commit/cdfcc4bfcc34ee348194dc9256a636a8fbb3491e

**Comment by dschuermann** [comment]
So it only fails on Android < 15 ?

**Comment by krt16s** [comment]
I tested my original solution on 15 and it worked. Just checked and 14 is okay as well, but I haven't tested it with 11/12/13. 10 and below crash without this fix.

**Comment by krt16s** [comment]
I opened https://github.com/SufficientlySecure/offline-calendar/pull/56 in favor of this one. Includes my previous solution with the fix for < 14 and defaults to light theme again. You can still access by old code in the master branch.

**Pull request closed without merging**